### ISSUE TITLE
changing only some rules

### ISSUE THREAD

**Issue by gagalago** [rrrene/credo#59]
Hello 
how can I change some rules without copy/paste all `.credo.exs` file ?
For example I would like disable `Credo.Check.Readability.ModuleDoc` and change `Credo.Check.Readability.MaxLineLength, priority: :low, max_length: 80` to 100. but I would like to keep all other rules.

**Comment by manukall** [comment]
I have the same question. I'd like to disable a rule without having to maintain the list of rules or forgetting to add a new rule when a new version is released.

**Comment by rrrene** [comment]
Hi there. Copypasta is currently the only option, but that behaviour will change soon. If sou have any suggestions/input on this it is more than welcome! 👍🏻

**Comment by smeevil** [comment]
How about a config file fallback chain that will look for a .credo.exs file in the current project, if that is not found then it will look for one in the home dir ~/.credo.exs , and if that is not found it will just use the defaults.

We could then format the custom config file for example like : 
```elixir
%{
  configs: [
    name: "default",
    ignore_checks: [
      {Credo.Check.Readability.ModuleDoc},
      ...
    ],
    override_checks: [
      {Credo.Check.Readability.MaxLineLength, priority: :low, max_length: 80},
      ...
    ]
  ]
}
 ```

as long as the custom config has no key 'checks' it will include all the checks as defined in the default config coming from credo, if it does have the key, it will naturally only use those checks. Basically we are using the default config, and then merge the settings from our custom config over it.

What do you think about this ?

**Comment by rrrene** [comment]
I could guess where you got the idea for this. :wink: Unfortunately, I am opposed to having multiple keys for this.

The way it will most probably work in `v0.4.0` is that the list of checks will be merged with the default list. The checks in your `.credo.exs` are prioritized and if you want to deactivate individual checks you just pust `{Credo.Check.CheckModule, false}` in the config.

P.S. The "config fallback" you describe is already active in Credo since - I think - the initial release. You can put a `.credo.exs` in any parent directory.

**Comment by smeevil** [comment]
Yeah the false annotation would work. 

Would I be correct in assuming that the config file will then only contain override rules ? 
So by default it would run all checks with default settings, unless in you specify otherwise in the config file. For example if you have  ```{Credo.Check.CheckModule, false}, {Credo.Check.Readability.MaxLineLength, priority: :low, max_length: 80}``` It would disable the CheckModule, override the default settings of the MaxLineLength and run all other checks with the defaults ?

 P.S. Good stuff with the config fallback :)

**Comment by rrrene** [comment]
Yepp, that's how I think it should work. 

Everyone I talk to tells me they don't want to monitor if new checks get implemented and that their prime use case is customizing/disabling a few checks. Quote: "Nobody wants to run just three checks."

So I am leaning very much towards following that line of thinking with the next "bigger" release.

**Comment by smeevil** [comment]
Sounds good :)

Let me know if there is anything that I could help with.

**Comment by rrrene** [comment]
This was released as part of `v0.4.0-beta1`. Please report any further issues you might find.
